src/common/no-deps.ts

```diff
@@ -0,0 +1,144 @@
+  logLevel: string
+  noInteractive: boolean
+  skipCleanup: boolean
+  trace: boolean
+  verbose: number
+}
+
+export const defaultBasicArguments: BasicArguments = {
+  _: [],
+  $0: 'defaultBasicArgs',
+  inDocker: true,
+  logLevel: 'WARN',
+  noInteractive: true,
+  skipCleanup: false,
+  trace: false,
+  verbose: 0,
+}
+
+let parsedArgs: BasicArguments
+
+export const setParsedArgs = (args: BasicArguments): void => {
+  parsedArgs = args
+}
+export const getParsedArgs = (): BasicArguments => {
+  return parsedArgs
+}
+
+export const asArray = (args: string | string[]): string[] => {
+  return Array.isArray(args) ? args : [args]
+}
+export const readdirRecurse = async (dir: string): Promise<string[]> => {
+  const dirs = readdirSync(dir, { withFileTypes: true })
+  const files = await Promise.all(
+    dirs.map(async (dirOrFile) => {
+      const res = resolve(dir, dirOrFile.name)
+      return dirOrFile.isDirectory() ? readdirRecurse(res) : res
+    }),
+  )
+  return files.flat()
+}
+
+export const capitalize = (s: string): string =>
+  (s &&
+    s
+      .split(' ')
+      .map((s2) => s2[0].toUpperCase() + s2.slice(1))
+      .join(' ')) ||
+  ''
+
+export const loadYaml = (path: string): any => {
+  if (!existsSync(path)) throw new Error(`${path} does not exists`)
+  return load(readFileSync(path, 'utf-8')) as any
+}
+
+export enum LOG_LEVELS {
+  FATAL = -2,
+  ERROR = -1,
+  WARN = 0,
+  INFO = 1,
+  DEBUG = 2,
+  TRACE = 3,
+}
+
+let logLevel = Number.NEGATIVE_INFINITY
+export const LOG_LEVEL = (): number => {
+  if (!getParsedArgs()) return LOG_LEVELS.ERROR
+  if (logLevel > Number.NEGATIVE_INFINITY) return logLevel
+
+  let logLevelNum = Number(LOG_LEVELS[getParsedArgs().logLevel?.toUpperCase() ?? 'WARN'])
+  const verbosity = Number(getParsedArgs().verbose ?? 0)
+  const boolTrace = env.TRACE || getParsedArgs().trace
+  logLevelNum = boolTrace ? LOG_LEVELS.TRACE : logLevelNum
+
+  logLevel = logLevelNum < 0 && verbosity === 0 ? logLevelNum : Math.max(logLevelNum, verbosity)
+  if (logLevel === LOG_LEVELS.TRACE) {
+    $.verbose = true
+    $.prefix = 'set -xeuo pipefail;'
+  }
+  return logLevel
+}
+
+export const LOG_LEVEL_STRING = (): string => {
+  return LOG_LEVELS[LOG_LEVEL()].toString()
+}
+
+// eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types
+export const deletePropertyPath = (object: any, path: string): void => {
+  if (!object || !path) {
+    return
+  }
+  const pathList = path.split('.')
+  let obj = object
+  for (let i = 0; i < pathList.length - 1; i++) {
+    obj = obj[pathList[i]]
+
+    if (!obj) {
+      return
+    }
+  }
+
+  delete obj[pathList.pop() as string]
+}
+export const delay = (ms: number): Promise<void> => new Promise((res) => setTimeout(res, ms))
+
+export const waitTillAvailable = async (domain: string, subsequentExists = 3): Promise<void> => {
+  let count = 0
+  // Need to wait for 3 subsequent exists, since DNS doesn't always propagate equally
+  do {
```

Comment:
In Otomi-tasks repo I have created faultTolerantFetch function some time ago.
It leverages `async-retry` lib. Maybe you could use the same approach.

Reply:
Doesn't seem to fully work (`faultTolerantFetch`)
https://github.com/redkubes/unassigned-issues/issues/236

That's why I created this function